server/STYLE.md

```diff
@@ -0,0 +1,505 @@
+# Code conventions and style guide
+
+This is a short document describing the preferred coding style for this project. We've tried to cover the major areas of formatting and naming. When something isn't covered by this guide, err on the side of consistency with existing code.
+
+### Formatting
+
+Use [stylish-haskell](https://github.com/jaspervdj/stylish-haskell) to format your code.
+
+#### Line Length
+
+Try to keep a line length within 100 characters. Small exceptions are okay if the line only goes over the limit by a few characters and breaking it would negatively impact readability. Use your best judgement.
+
+#### Whitespace
+
+Do not use tabs. Use spaces for indenting. Indent your code blocks with **2
+spaces**. Indent the `where` keyword two spaces to set it apart from the rest of
+the code and indent the definitions in a `where` clause 2 spaces. Some examples:
+
+```haskell
+sayHello :: IO ()
+sayHello = do
+  name <- getLine
+  putStrLn $ greeting name
+  where
+    greeting name = "Hello, " ++ name ++ "!"
+
+filter :: (a -> Bool) -> [a] -> [a]
+filter _ []   = []
+filter p (x:xs)
+  | p x       = x : filter p xs
+  | otherwise = filter p xs
+```
+
+Include newlines at the ends of files. Do not include other trailing whitespace on the end of lines.
+
+#### Data Declarations
+
+Align the constructors in a data type definition. Example:
+
+```haskell
+data Tree a
+  = Branch !a !(Tree a) !(Tree a)
+  | Leaf
+```
+
+```haskell
+data HttpException
+  = InvalidStatusCode Int
+  | MissingContentHeader
+```
+
+Format records as follows:
+
+```haskell
+data Person = Person
+  { firstName :: !String  -- ^ First name
+  , lastName  :: !String  -- ^ Last name
+  , age       :: !Int     -- ^ Age
+  } deriving (Eq, Show)
+```
+
+#### List Declarations
+
+Align the elements in the list.  Example:
+
+```haskell
+exceptions =
+  [ InvalidStatusCode
+  , MissingContentHeader
+  , InternalServerError
+  ]
+```
+
+Optionally, you can skip the first newline.  Use your judgement.
+
+```haskell
+directions = [ North
+             , East
+             , South
+             , West
+             ]
+```
+
+#### Export Lists
+
+Format export lists as follows:
+
+```haskell
+module Data.Set
+  ( -- * The @Set@ type
+    Set
+  , empty
+  , singleton
+
+    -- * Querying
+  , member
+  ) where
+```
+
+### Imports
+
+Imports should be grouped in the following order:
+
+1. `Hasura.Prelude`
+2. `qualified` imports of standard library / third party imports
+3. unqualified standard library imports / related third party imports
+4. `qualified` imports of local application/library specific imports
```

Comment:
Do we have specific guidelines on how imports should be abbreviated?
- import qualified Data.Map as Map
- import qualified Data.Map as M

And do we have specific guidelines between qualified / unqualified imports, with or without explicit import lists?

Reply:
@nicuveo We used to have `as M` previously, but currently we use more readable names like Map, Json, Set etc. 

> And do we have specific guidelines between qualified / unqualified imports, with or without explicit import lists?

I don't think we have a guideline for but the existing style is to use a qualified import for external libraries and we import frequently used internal modules directly.

Reply:
An older version of this style guide recommended using either qualified imports or explicit import lists for external libraries, but I removed that sentence as part of the cleanup, since it didn’t seem like we were actually following that guideline, and I’m personally fine with using unqualified imports without an import list. ¯\_(ツ)_/¯

If someone feels strongly about it, by all means, make a case for it—I would consider adding it back! But I’d rather not codify things in the style guide unless either (a) we can automatically enforce them or (b) we all agree on them enough to enforce them ourselves. Otherwise, adding them to the style guide seems unlikely to do much.